packages/web/src/pages/user/[uid].tsx

```diff
@@ -38,22 +41,47 @@ const UserProfilePage: NextPage = () => {
     fetchUser();
   }, [uid]);
 
+  React.useEffect(() => {
+    const checkUser = async () => {
+      try {
+        const res = await fetch('/api/users/me');
+
+        if (res.ok) {
+          const data = await res.json();
+
+          if (user && data.id == user.id) {
+            setIsCurrentUser(true);
+          }
+        }
+        setDidCheckCurrentUser(true);
+      } catch {}
+    };
+
+    checkUser();
+  }, [user]);
+
   if (user === undefined) {
     if (errorMessage == '') return null;
     else
       return (
-        <MarketingLayout>
+        <AppLayout>
           <Alert status="error">
             <AlertIcon />
             {errorMessage}
           </Alert>
-        </MarketingLayout>
+        </AppLayout>
       );
   } else {
     return (
-      <MarketingLayout>
-        <UserProfile user={user} setUser={setUser} />
-      </MarketingLayout>
+      <AppLayout>
+        {!didCheckCurrentUser ? (
```

Comment:
This is a really ambiguous variable name and I had no idea what it was doing when saw it being used up top. Can we rename this to be more meaningful? Something like `error` should be more readable.